demisto_sdk/commands/validate/validators/RN_validators/RN111_is_docker_entry_match_yml.py

```diff
@@ -4,60 +4,127 @@
+                        docker_entry.find("*") + 1 : docker_entry.rfind("*")
+                    ]
+                    break
+    return docker
+
+
+def release_notes_mismatch_error(content_item: IntegrationScript):
+    should_be_entry = release_notes_shouldbe_entry(content_item)
+    image_entry = get_docker_image_entry(
+        content_item.pack.release_note.file_content, content_item.name
+    )
+    if should_be_entry and (
+        should_be_entry not in image_entry or image_entry == NO_DOCKER_ENTRY_FOUND
+    ):
+        return f"Docker version in release notes should be {should_be_entry}, found: {image_entry}"
+    if not should_be_entry and image_entry and not image_entry == NO_DOCKER_ENTRY_FOUND:
+        return f"There should be no release notes docker update entry, found: {image_entry}"
+
+
 class IsDockerEntryMatchYmlValidator(BaseValidator[ContentTypes]):
     error_code = "RN111"
     description = "Validate that the docker image version mentioned in the RN is indeed the one in the mentioned in the yml file."
     rationale = "We want to make sure we don't document wrong information."
-    error_message = "The docker entry in the release notes doesn't match what is in the yml.\n The docker image in rn: {0}, docker image in yml {1} - please make sure the dockers match."
+    error_message = "The release notes regarding the docker image are not correct. {0}"
     related_field = "docker_image"
     expected_git_statuses = [GitStatuses.MODIFIED, GitStatuses.RENAMED]
     related_file_type = [RelatedFileType.RELEASE_NOTE]
+    is_auto_fixable = True
 
     def obtain_invalid_content_items(
         self, content_items: Iterable[ContentTypes]
     ) -> List[ValidationResult]:
         return [
             ValidationResult(
                 validator=self,
-                message=self.error_message.format(
-                    docker_entry, content_item.docker_image
-                ),
+                message=self.error_message.format(error),
                 content_object=content_item,
             )
             for content_item in content_items
-            if (old_obj := content_item.old_base_content_object)
-            and content_item.docker_image != old_obj.docker_image  # type:ignore[attr-defined]
-            and (
-                docker_entry := self.get_docker_image_entry(
-                    content_item.pack.release_note.file_content,
-                    content_item.display_name,
-                )
-            )
-            and content_item.docker_image not in docker_entry
+            if (error := release_notes_mismatch_error(content_item))
         ]
 
-    def get_docker_image_entry(self, rn: str, content_item_name: str) -> str:
-        rn_items = rn.split("##### ")
-        docker = "No docker entry found"
+    def fix(
+        self,
+        content_item: ContentTypes,
+    ) -> FixResult:
+        should_be_rn_entry = release_notes_shouldbe_entry(content_item)
```

Comment:
why not keeping the results from the validation part in a class var and using it here rather than running it again? see IN102 as reference

Reply:
i think its easier to just add @cached on this method, wdyt